### PR TITLE
Delete unused System.Runtime.Versioning.CompatibilitySwitch

### PR DESCRIPTION
**Pull request by huoyaoyuan**
It's only used in XplatEventLogger. Updated to read the environment variable directly.

A concern is prefix rule being duplicated.

### CODE REVIEW

**Comment by tarekgh** [comment]
I am not sure how important this, doesn't `CLRConfig::GetConfigValue` support reading the config from other sources other than the environment variables? 

CC @jkotas

**Comment by jkotas** [comment]
> A concern is prefix rule being duplicated.

Would it look better to convert it to QCall that lives next to the other XplatEventLogger QCalls? I agree that this does not need to be a general purpose API given that it is only used by `XplatEventLogger`.

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/XplatEventLogger.cs [hunk 1]
```diff
@@ -18,8 +18,13 @@ internal sealed partial class XplatEventLogger : EventListener
     {
         public XplatEventLogger() { }
 
-        private static readonly Lazy<string?> eventSourceNameFilter = new Lazy<string?>(() => CompatibilitySwitch.GetValueInternal("EventSourceFilter"));
-        private static readonly Lazy<string?> eventSourceEventFilter = new Lazy<string?>(() => CompatibilitySwitch.GetValueInternal("EventNameFilter"));
+        private static readonly Lazy<string?> eventSourceNameFilter = new Lazy<string?>(() => GetClrConfig("EventSourceFilter"));
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/XplatEventLogger.cs:21]: You can drop `Lazy` while you are on it. The allocation and initialization of `Lazy<T>` costs about as much as just fetching the config string.

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/XplatEventLogger.cs [hunk 2]
```diff
@@ -18,8 +15,13 @@ internal sealed partial class XplatEventLogger : EventListener
     {
         public XplatEventLogger() { }
 
-        private static readonly Lazy<string?> eventSourceNameFilter = new Lazy<string?>(() => CompatibilitySwitch.GetValueInternal("EventSourceFilter"));
-        private static readonly Lazy<string?> eventSourceEventFilter = new Lazy<string?>(() => CompatibilitySwitch.GetValueInternal("EventNameFilter"));
+        private static readonly string eventSourceNameFilter = GetClrConfig("EventSourceFilter");
+        private static readonly string eventSourceEventFilter = GetClrConfig("EventNameFilter");
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/XplatEventLogger.cs:19]:
```suggestion
        private static readonly string s_eventSourceNameFilter = GetClrConfig("EventSourceFilter");
        private static readonly string s_eventSourceEventFilter = GetClrConfig("EventNameFilter");
```
Naming convention

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/XplatEventLogger.cs [hunk 3]
```diff
@@ -158,9 +155,7 @@ private static void AppendByteArrayAsHexString(ref ValueStringBuilder builder, b
 
         protected internal override void OnEventSourceCreated(EventSource eventSource)
         {
-
-            string? eventSourceFilter = eventSourceNameFilter.Value;
-            if (string.IsNullOrEmpty(eventSourceFilter) || (eventSource.Name.Contains(eventSourceFilter, StringComparison.OrdinalIgnoreCase)))
+            if (string.IsNullOrEmpty(eventSourceNameFilter) || (eventSource.Name.Contains(eventSourceNameFilter, StringComparison.OrdinalIgnoreCase)))
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/XplatEventLogger.cs:158]:
```suggestion
            if (string.IsNullOrEmpty(s_eventSourceNameFilter) || (eventSource.Name.Contains(s_eventSourceNameFilter, StringComparison.OrdinalIgnoreCase)))
```

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/XplatEventLogger.cs [hunk 4]
```diff
@@ -178,8 +173,7 @@ protected internal override void OnEventWritten(EventWrittenEventArgs eventData)
                 return;
             }
 
-            string? eventFilter = eventSourceEventFilter.Value;
-            if (string.IsNullOrEmpty(eventFilter) || (eventData.EventName!.Contains(eventFilter, StringComparison.OrdinalIgnoreCase)))
+            if (string.IsNullOrEmpty(eventSourceEventFilter) || (eventData.EventName!.Contains(eventSourceEventFilter, StringComparison.OrdinalIgnoreCase)))
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/XplatEventLogger.cs:176]:
```suggestion
            if (string.IsNullOrEmpty(s_eventSourceEventFilter) || (eventData.EventName!.Contains(s_eventSourceEventFilter, StringComparison.OrdinalIgnoreCase)))
```

**Review by jkotas — APPROVED** [review]
Thanks!

**Pull request merged**